src/server/error-controller.js

```diff
@@ -70,17 +70,17 @@ class ErrorController {
   }
 
   tryRespondToExpiredSession(req, res, err) {
-    if (!req.isAuthenticated() && req.cookies[this.serverConfig.sessionCookieName]) {
+    if (err.status === HTTP_STATUS.unauthorized && !req.isAuthenticated() && req.cookies[this.serverConfig.sessionCookieName]) {
```

Comment:
It happened to me while testing that I got a "session expired" instead of the real underlying error. It should only assume that the user wants to re-login because of an expired session when there was a 401 before.